a-list.js

```diff
@@ -0,0 +1,33 @@
+function arrayToList(array){
+  let entry = null;
+  for (let i = array.length-1; i >= 0; i--){
+    entry = {value: array[i], rest: entry};
```

Comment:
same here

Reply:
This is looping through the array backwards, starting with the last item (`array.length-1`) and subtract one from `i` each time until `i = 0` which is `array[0]` the first item in the array.

The variable `entry` is being reassigned each time, and the value of the `rest` key for each object is the previous entry, like Russian nesting dolls.